pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -850,6 +850,17 @@ The delayed message index time step(in seconds) in per bucket snapshot segment,
         doc = "Enable check for minimum allowed client library version"
     )
     private boolean clientLibraryVersionCheckEnabled = false;
+
+    @FieldContext(
+        category = CATEGORY_SERVER,
+        dynamic = true,
+        doc = "Minimum client version allowed by broker else broker will reject connection."
+              + "(It's useful when client lib doesn't support specific feature and feature "
+              + "might be required by broker to apply globally on all topics."
+              + "(eg: all clients must be on V20 to perform cloud migration)"
+    )
+    private int clientMinVersionAllowed = -1;
```

Comment:
Would it make sense to add proto or protocol to the name to disambiguate it from the library version?